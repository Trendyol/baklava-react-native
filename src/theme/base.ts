export const space = [0, 2, 4, 8, 12, 16, 24, 32, 48, 64];

export const fonts = ['System', 'System', 'System', 'System', 'System'];

export const fontSizes = [12, 14, 16, 20, 24, 28, 32];

export const fontWeights = ['300', '400', '500', '600', '700'];

export const lineHeights = [12, 14, 16, 20, 24, 28, 32, 36];

export const radii = {
  none: 0,
  small: 4,
  normal: 6,
  large: 8,
  xlarge: 16,
};

export const colors = {
  white: '#ffffff',
  black: '#000000',
  transparent: 'transparent',
  // Color
  primaryColor: '#F27A1A',
  secondaryColor: '#273142',
  tertiaryColor: '#F1F2F7',
  successColor: '#0BC15C',
  dangerColor: '#FF5043',
  warningColor: '#FFB600',
  alternativeColor: '#5794FF',
  featuredColor: '#8C4EFF',
  // Hover
  primaryHover: '#EF6114',
  secondaryHover: '#0F131A',
  tertiaryHover: '#D5D9E1',
  successHover: '#09A44E',
  dangerHover: '#FF3028',
  warningHover: '#FF9800',
  alternativeHover: '#457EFF',
  featuredHover: '#753EFF',
  // Background
  primaryBackground: '#F1F2F7',
  accentBackground: '#FEF2E8',
  secondaryBackground: '#FFFFFF',
  tertiaryBackground: '#F7F7FA',
  successBackground: '#E7F9EF',
  alertBackground: '#FFEEEC',
  warningBackground: '#FFF8E6',
  infoBackground: '#EEF4FF',
  featuredCBackground: '#F4EDFF',
  // Border Color
  borderColor: '#D5D9E1',
  // Content
  contentPrimary: '#273142',
  contentSecondary: '#6E7787',
  contentTertiary: '#95A1B5',
  contentPassive: '#AFBBCA',
};

export default {
  space,
  fonts,
  fontSizes,
  fontWeights,
  lineHeights,
  radii,
  colors,
};
