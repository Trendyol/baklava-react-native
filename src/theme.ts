import { createTheme } from '@shopify/restyle';

const theme = createTheme({
  colors: {
    white: '#ffffff',
    black: '#000000',
    transparent: 'transparent',
    // Color
    primaryColor: '#F27A1A',
    secondaryColor: '#273142',
    tertiaryColor: '#F1F2F7',
    successColor: '#0BC15C',
    dangerColor: '#FF5043',
    warningColor: '#FFB600',
    alternativeColor: '#5794FF',
    featuredColor: '#8C4EFF',
    // Hover
    primaryHover: '#EF6114',
    secondaryHover: '#0F131A',
    tertiaryHover: '#D5D9E1',
    successHover: '#09A44E',
    dangerHover: '#FF3028',
    warningHover: '#FF9800',
    alternativeHover: '#457EFF',
    featuredHover: '#753EFF',
    // Background
    primaryBackground: '#F1F2F7',
    accentBackground: '#FEF2E8',
    secondaryBackground: '#FFFFFF',
    tertiaryBackground: '#F7F7FA',
    successBackground: '#E7F9EF',
    alertBackground: '#FFEEEC',
    warningBackground: '#FFF8E6',
    infoBackground: '#EEF4FF',
    featuredCBackground: '#F4EDFF',
    // Border Color
    borderColor: '#D5D9E1',
    // Content
    contentPrimary: '#273142',
    contentSecondary: '#6E7787',
    contentTertiary: '#95A1B5',
    contentPassive: '#AFBBCA',
  },
  spacing: {
    none: 0,
    '4xs': 2,
    '3xs': 4,
    '2xs': 8,
    xs: 12,
    s: 14,
    m: 16,
    l: 20,
    xl: 24,
    '2xl': 32,
    '3xl': 40,
    '4xl': 48,
    '5xl': 64,
    '6xl': 80,
  },
  borderRadii: {
    none: 0,
    xs: 2,
    s: 4,
    m: 6,
    l: 8,
    xl: 16,
  },
  breakpoints: {
    phone: 0,
    tablet: 768,
  },
  fonts: {
    light: 'System',
    regular: 'System',
    medium: 'System',
    semiBold: 'System',
    bold: 'System',
  },
  textVariants: {
    defaults: {
      fontSize: 14,
      fontWeight: '400',
      fontFamily: 'regular',
      lineHeight: 16,
      color: 'contentPrimary',
    },
    // Heading
    heading1: {
      fontSize: 32,
      fontWeight: '400',
      fontFamily: 'regular',
      lineHeight: 36,
    },
    heading2: {
      fontSize: 28,
      fontWeight: '400',
      fontFamily: 'regular',
      lineHeight: 32,
    },
    heading3: {
      fontSize: 24,
      fontWeight: '400',
      fontFamily: 'regular',
      lineHeight: 28,
    },
    // Subtitle01
    subtitle01Regular: {
      fontSize: 20,
      fontWeight: '400',
      fontFamily: 'regular',
      lineHeight: 24,
    },
    subtitle01Medium: {
      fontSize: 20,
      fontWeight: '500',
      fontFamily: 'medium',
      lineHeight: 24,
    },
    subtitle01Semibold: {
      fontSize: 20,
      fontWeight: '600',
      fontFamily: 'semiBold',
      lineHeight: 24,
    },
    subtitle01Bold: {
      fontSize: 20,
      fontWeight: '700',
      fontFamily: 'bold',
      lineHeight: 24,
    },
    // Subtitle02
    subtitle02Regular: {
      fontSize: 16,
      fontWeight: '400',
      fontFamily: 'regular',
      lineHeight: 20,
    },
    subtitle02Medium: {
      fontSize: 16,
      fontWeight: '500',
      fontFamily: 'medium',
      lineHeight: 20,
    },
    subtitle02Semibold: {
      fontSize: 16,
      fontWeight: '600',
      fontFamily: 'semiBold',
      lineHeight: 20,
    },
    subtitle02Bold: {
      fontSize: 16,
      fontWeight: '700',
      fontFamily: 'bold',
      lineHeight: 20,
    },
    // Subtitle03
    subtitle03Regular: {
      fontSize: 14,
      fontWeight: '400',
      fontFamily: 'regular',
      lineHeight: 16,
    },
    subtitle03Medium: {
      fontSize: 14,
      fontWeight: '500',
      fontFamily: 'medium',
      lineHeight: 16,
    },
    subtitle03Semibold: {
      fontSize: 14,
      fontWeight: '600',
      fontFamily: 'semiBold',
      lineHeight: 16,
    },
    subtitle03Bold: {
      fontSize: 14,
      fontWeight: '700',
      fontFamily: 'bold',
      lineHeight: 16,
    },
    // Subtitle04
    subtitle04Regular: {
      fontSize: 12,
      fontWeight: '400',
      fontFamily: 'regular',
      lineHeight: 14,
    },
    subtitle04Medium: {
      fontSize: 12,
      fontWeight: '500',
      fontFamily: 'medium',
      lineHeight: 14,
    },
    subtitle04Semibold: {
      fontSize: 12,
      fontWeight: '600',
      fontFamily: 'semiBold',
      lineHeight: 14,
    },
    subtitle04Bold: {
      fontSize: 12,
      fontWeight: '700',
      fontFamily: 'bold',
      lineHeight: 14,
    },
    // Body
    bodyText: {
      fontSize: 14,
      fontWeight: '400',
      fontFamily: 'regular',
      lineHeight: 16,
    },
    bodyUnderline: {
      fontSize: 14,
      fontWeight: '400',
      fontFamily: 'regular',
      lineHeight: 16,
      textDecorationLine: 'underline',
    },
    bodyTextLink: {
      fontSize: 14,
      fontWeight: '500',
      fontFamily: 'medium',
      lineHeight: 16,
      textDecorationLine: 'underline',
    },
    bodyLongText: {
      fontSize: 14,
      fontWeight: '400',
      fontFamily: 'regular',
      lineHeight: 16,
    },
    // Caption
    captionText: {
      fontSize: 12,
      fontWeight: '400',
      fontFamily: 'regular',
      lineHeight: 12,
    },
    captionMedium: {
      fontSize: 12,
      fontWeight: '500',
      fontFamily: 'medium',
      lineHeight: 12,
    },
    captionLongText: {
      fontSize: 12,
      fontWeight: '400',
      fontFamily: 'regular',
      lineHeight: 12,
    },
    captionTextLink: {
      fontSize: 12,
      fontWeight: '500',
      fontFamily: 'medium',
      lineHeight: 12,
      textDecorationLine: 'underline',
    },
  },
});

export type Theme = typeof theme;
export default theme;
