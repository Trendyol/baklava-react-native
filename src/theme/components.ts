import { variant } from 'styled-system';
import {
  ButtonTypeTypes,
  ButtonVariantTypes,
} from '../components/Button/types';

// TODO: Unitless shorthands https://github.com/styled-components/css-to-react-native/issues/40
export const textStyle = {
  // Heading
  heading1: {
    fontSize: 6,
    fontWeight: 1,
    fontFamily: 1,
    // lineHeight: 7,
  },
  heading2: {
    fontSize: 5,
    fontWeight: 1,
    fontFamily: 1,
    // lineHeight: 6,
  },
  heading3: {
    fontSize: 4,
    fontWeight: 1,
    fontFamily: 1,
    // lineHeight: 5,
  },
  // Subtitle01
  subtitle01Regular: {
    fontSize: 3,
    fontWeight: 1,
    fontFamily: 1,
    // lineHeight: 4,
  },
  subtitle01Medium: {
    fontSize: 3,
    fontWeight: 2,
    fontFamily: 2,
    // lineHeight: 4,
  },
  subtitle01Semibold: {
    fontSize: 3,
    fontWeight: 3,
    fontFamily: 3,
    // lineHeight: 4,
  },
  subtitle01Bold: {
    fontSize: 3,
    fontWeight: 4,
    fontFamily: 4,
    // lineHeight: 4,
  },
  // Subtitle02
  subtitle02Regular: {
    fontSize: 2,
    fontWeight: 1,
    fontFamily: 1,
    // lineHeight: 3,
  },
  subtitle02Medium: {
    fontSize: 2,
    fontWeight: 2,
    fontFamily: 2,
    // lineHeight: 3,
  },
  subtitle02Semibold: {
    fontSize: 2,
    fontWeight: 3,
    fontFamily: 3,
    // lineHeight: 3,
  },
  subtitle02Bold: {
    fontSize: 2,
    fontWeight: 4,
    fontFamily: 4,
    // lineHeight: 3,
  },
  // Subtitle03
  subtitle03Regular: {
    fontSize: 1,
    fontWeight: 1,
    fontFamily: 1,
    // lineHeight: 2,
  },
  subtitle03Medium: {
    fontSize: 1,
    fontWeight: 2,
    fontFamily: 2,
    // lineHeight: 2,
  },
  subtitle03Semibold: {
    fontSize: 1,
    fontWeight: 3,
    fontFamily: 3,
    // lineHeight: 2,
  },
  subtitle03Bold: {
    fontSize: 1,
    fontWeight: 4,
    fontFamily: 4,
    // lineHeight: 2,
  },
  // Subtitle04
  subtitle04Regular: {
    fontSize: 0,
    fontWeight: 1,
    fontFamily: 1,
    // lineHeight: 1,
  },
  subtitle04Medium: {
    fontSize: 0,
    fontWeight: 2,
    fontFamily: 2,
    // lineHeight: 1,
  },
  subtitle04Semibold: {
    fontSize: 0,
    fontWeight: 3,
    fontFamily: 3,
    // lineHeight: 1,
  },
  subtitle04Bold: {
    fontSize: 0,
    fontWeight: 4,
    fontFamily: 4,
    // lineHeight: 1,
  },
  // Body
  bodyText: {
    fontSize: 1,
    fontWeight: 1,
    fontFamily: 1,
    // lineHeight: 2,
  },
  bodyUnderline: {
    fontSize: 1,
    fontWeight: 1,
    fontFamily: 1,
    // lineHeight: 2,
    textDecorationLine: 'underline',
  },
  bodyTextLink: {
    fontSize: 1,
    fontWeight: 2,
    fontFamily: 2,
    // lineHeight: 2,
    textDecorationLine: 'underline',
  },
  bodyLongText: {
    fontSize: 1,
    fontWeight: 1,
    fontFamily: 1,
    // lineHeight: 2,
  },
  // Caption
  captionText: {
    fontSize: 0,
    fontWeight: 1,
    fontFamily: 1,
    // lineHeight: 0,
  },
  captionMedium: {
    fontSize: 0,
    fontWeight: 2,
    fontFamily: 2,
    // lineHeight: 0,
  },
  captionLongText: {
    fontSize: 0,
    fontWeight: 1,
    fontFamily: 1,
    // lineHeight: 0,
  },
  captionTextLink: {
    fontSize: 0,
    fontWeight: 2,
    fontFamily: 2,
    // lineHeight: 0,
    textDecorationLine: 'underline',
  },
};

const buttonCommon = {
  borderRadius: 'normal',
};

export const buttonStyles = {
  height: [32, 40, 48],
  padding: [8, 12, 14, 16, 24, 32],
  variant: {
    primary: {
      color: 'white',
      backgroundColor: 'primaryColor',
      ...buttonCommon,
    },
    secondary: {
      color: 'white',
      backgroundColor: 'secondaryColor',
      ...buttonCommon,
    },
    success: {
      color: 'white',
      backgroundColor: 'successColor',
      ...buttonCommon,
    },
    danger: {
      color: 'white',
      backgroundColor: 'dangerColor',
      ...buttonCommon,
    },
  },
  type: (props: {
    variant: ButtonVariantTypes;
    isPressed: boolean;
    disabled: boolean;
    type: ButtonTypeTypes;
  }) => {
    const variantColor = props.disabled
      ? 'tertiaryColor'
      : `${props.variant}Color` ?? 'primaryBackground';
    const variantHoverColor = `${props.variant}Hover` ?? 'primaryBackground';

    return {
      contained: {
        color: props.disabled ? 'secondaryColor' : 'white',
        backgroundColor: props.disabled
          ? 'tertiaryColor'
          : props.isPressed
          ? variantHoverColor
          : variantColor,
        borderWidth: 1,
        borderColor: variantColor,
        ...buttonCommon,
      },
      outline: {
        color: props.isPressed ? 'white' : variantColor,
        backgroundColor: props.disabled
          ? variantColor
          : props.isPressed
          ? variantHoverColor
          : 'transparent',
        borderWidth: 1,
        borderColor: variantColor,
        ...buttonCommon,
      },
      text: {
        color: props.isPressed ? variantHoverColor : variantColor,
        backgroundColor: 'transparent',
        borderWidth: 1,
        borderColor: 'transparent',
        textDecorationLine: 'underline',
        ...buttonCommon,
      },
    };
  },
  size: (props: { text: string }) => {
    return {
      large: {
        paddingVertical: buttonStyles.padding[2],
        paddingHorizontal: props.text
          ? buttonStyles.padding[5]
          : buttonStyles.padding[2],
      },
      medium: {
        paddingVertical: buttonStyles.padding[1],
        paddingHorizontal: props.text
          ? buttonStyles.padding[4]
          : buttonStyles.padding[1],
      },
      small: {
        paddingVertical: buttonStyles.padding[0],
        paddingHorizontal: props.text
          ? buttonStyles.padding[3]
          : buttonStyles.padding[0],
      },
    };
  },
};

export const buttonVariant = variant({
  prop: 'variant',
  variants: buttonStyles.variant,
});

export const buttonType = (props: any) =>
  variant({
    prop: 'type',
    variants: buttonStyles.type(props),
  });

export const buttonSize = (props: any) =>
  variant({
    prop: 'size',
    variants: buttonStyles.size(props),
  });

export const inputStyle = {
  height: [32, 40, 48],
};

export const iconSizes = [12, 14, 16, 20, 24, 32];

export default {
  textStyle,
  buttonStyles,
  buttonVariant,
  buttonType,
  buttonSize,
  inputStyle,
  iconSizes,
};
