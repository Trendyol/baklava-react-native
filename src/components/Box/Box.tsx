import { createBox } from '@ergenekonyigit/restyle';
import { Theme } from '../../theme';

const Box = createBox<Theme>();

export default Box;
