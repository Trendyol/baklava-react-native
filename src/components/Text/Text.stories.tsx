import React from 'react';
import { ComponentStory, ComponentMeta } from '@storybook/react-native';
import Text from './Text';
import theme from '../../theme';

const variantList = Object.keys(theme.textVariants);

const TextMeta: ComponentMeta<typeof Text> = {
  title: 'Text',
  component: Text,
  argTypes: {
    variant: {
      options: variantList,
      control: { type: 'radio' },
    },
  },
  args: {
    variant: variantList[0] as any,
  },
};

export default TextMeta;

type TextStory = ComponentStory<typeof Text>;

export const Basic: TextStory = args => (
  <>
    <Text p="2xs" variant="subtitle1Bold">
      Text
    </Text>
    <Text variant={args.variant} p="2xs">
      Sample Text
    </Text>
  </>
);

export const Heading: TextStory = () => (
  <>
    <Text p="2xs" variant="subtitle1Bold">
      Text Heading
    </Text>
    <Text variant="heading1" p="2xs">
      Heading 1
    </Text>
    <Text variant="heading2" p="2xs">
      Heading 2
    </Text>
    <Text variant="heading3" p="2xs">
      Heading 3
    </Text>
  </>
);

export const Subtitle1: TextStory = () => (
  <>
    <Text p="2xs" variant="subtitle1Bold">
      Text Subtitle 1
    </Text>
    <Text variant="subtitle1Regular" p="2xs">
      Subtitle 1 / Regular
    </Text>
    <Text variant="subtitle1Medium" p="2xs">
      Subtitle 1 / Medium
    </Text>
    <Text variant="subtitle1Semibold" p="2xs">
      Subtitle 1 / Semibold
    </Text>
    <Text variant="subtitle1Bold" p="2xs">
      Subtitle 1 / Bold
    </Text>
  </>
);

export const Subtitle2: TextStory = () => (
  <>
    <Text p="2xs" variant="subtitle1Bold">
      Text Subtitle 2
    </Text>
    <Text variant="subtitle2Regular" p="2xs">
      Subtitle 2 / Regular
    </Text>
    <Text variant="subtitle2Medium" p="2xs">
      Subtitle 2 / Medium
    </Text>
    <Text variant="subtitle2Semibold" p="2xs">
      Subtitle 2 / Semibold
    </Text>
    <Text variant="subtitle2Bold" p="2xs">
      Subtitle 2 / Bold
    </Text>
  </>
);

export const Subtitle3: TextStory = () => (
  <>
    <Text p="2xs" variant="subtitle1Bold">
      Text Subtitle 3
    </Text>
    <Text variant="subtitle3Regular" p="2xs">
      Subtitle 3 / Regular
    </Text>
    <Text variant="subtitle3Medium" p="2xs">
      Subtitle 3 / Medium
    </Text>
    <Text variant="subtitle3Semibold" p="2xs">
      Subtitle 3 / Semibold
    </Text>
    <Text variant="subtitle3Bold" p="2xs">
      Subtitle 3 / Bold
    </Text>
  </>
);

export const Subtitle4: TextStory = () => (
  <>
    <Text p="2xs" variant="subtitle1Bold">
      Text Subtitle 4
    </Text>
    <Text variant="subtitle4Regular" p="2xs">
      Subtitle 4 / Regular
    </Text>
    <Text variant="subtitle4Medium" p="2xs">
      Subtitle 4 / Medium
    </Text>
    <Text variant="subtitle4Semibold" p="2xs">
      Subtitle 4 / Semibold
    </Text>
    <Text variant="subtitle4Bold" p="2xs">
      Subtitle 4 / Bold
    </Text>
  </>
);

export const Body: TextStory = () => (
  <>
    <Text p="2xs" variant="subtitle1Bold">
      Text Body
    </Text>
    <Text variant="body1" p="2xs">
      Body 1
    </Text>
    <Text variant="body2" p="2xs">
      Body 2
    </Text>
    <Text variant="body3" p="2xs">
      Body 3
    </Text>
    {/* <Text variant="bodyUnderline" p="2xs">
      Body / Underline
    </Text>
    <Text variant="bodyTextLink" p="2xs">
      Body / Text Link
    </Text>
    <Text variant="bodyLongText" p="2xs">
      Body / Long Text
    </Text> */}
  </>
);

export const Caption: TextStory = () => (
  <>
    <Text p="2xs" variant="subtitle1Bold">
      Text Caption
    </Text>
    <Text variant="captionText" p="2xs">
      Caption / Text
    </Text>
  </>
);
