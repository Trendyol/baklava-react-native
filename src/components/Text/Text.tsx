import {
  composeRestyleFunctions,
  TextProps,
  useRestyle,
  color,
  createVariant,
  opacity,
  spacing,
  spacingShorthand,
  textShadow,
  typography,
  visible,
} from '@ergenekonyigit/restyle';
import React from 'react';
import { TextProps as RNTextProps, Text as RNText } from 'react-native';
import { Theme } from '../../theme';

export type TextVariants = TextProps<Theme>['variant'];

type RestyleProps = TextProps<Theme>;

const restyleFunctions = composeRestyleFunctions<Theme, RestyleProps>([
  color,
  opacity,
  visible,
  typography,
  spacing,
  spacingShorthand,
  textShadow,
  createVariant({ themeKey: 'textVariants' }),
]);

type Props = RNTextProps & RestyleProps;

const Text = ({ testID, accessibilityLabel, accessible, ...rest }: Props) => {
  const props = useRestyle(restyleFunctions, rest);

  const testProps = React.useMemo(() => {
    const result: RNTextProps = {
      accessible: false,
      accessibilityLabel: undefined,
      testID: undefined,
    };

    if (!(testID || accessibilityLabel)) {
      return result;
    }

    result.accessible = accessible ?? true;
    result.accessibilityLabel = accessibilityLabel;
    result.testID = testID ?? accessibilityLabel;

    return result;
  }, [testID, accessibilityLabel, accessible]);

  return <RNText {...props} {...testProps} />;
};

export default Text;
