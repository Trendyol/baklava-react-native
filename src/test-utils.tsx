import { ThemeProvider } from '@ergenekonyigit/restyle';
import { render } from '@testing-library/react-native';
import * as React from 'react';
import {
  GestureResponderHandlers,
  PanResponder,
  PanResponderCallbacks,
} from 'react-native';
import theme from './theme';
import { PortalProvider } from '.';

theme.fonts = {
  light: 'Rubik-Light',
  regular: 'Rubik-Regular',
  medium: 'Rubik-Medium',
  semiBold: 'Rubik-SemiBold',
  bold: 'Rubik-Bold',
};

export interface RenderOptions {
  wrapper?: React.ComponentType<any>;
  createNodeMock?: (element: React.ReactElement<any>) => any;
}

const AllProviders = ({ children }: { children?: React.ReactNode }) => (
  <ThemeProvider theme={theme}>
    <PortalProvider>{children}</PortalProvider>
  </ThemeProvider>
);

const customRender = (ui: React.ReactElement, options?: RenderOptions) =>
  render(ui, { wrapper: AllProviders, ...options }) as any;

const mockPanResponder = () => {
  jest
    .spyOn(PanResponder, 'create')
    .mockImplementation(
      ({
        onStartShouldSetPanResponderCapture,
        onMoveShouldSetPanResponder,
        onPanResponderGrant,
        onPanResponderMove,
        onPanResponderRelease,
      }: PanResponderCallbacks) => ({
        panHandlers: {
          onStartShouldSetResponderCapture: onStartShouldSetPanResponderCapture,
          onMoveShouldSetResponder: onMoveShouldSetPanResponder,
          onResponderGrant: onPanResponderGrant,
          onResponderMove: onPanResponderMove,
          onResponderRelease: onPanResponderRelease,
        } as GestureResponderHandlers,
      }),
    );
};

export * from '@testing-library/react-native';
export { customRender as render, mockPanResponder };
