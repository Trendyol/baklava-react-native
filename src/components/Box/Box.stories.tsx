import { ComponentStory } from '@storybook/react-native';
import React from 'react';
import Text from '../Text/Text';
import Box from './Box';

export default {
  title: 'Box',
  component: Box,
};

type BoxStory = ComponentStory<typeof Box>;

export const Basic: BoxStory = args => (
  <>
    <Text p="2xs" variant="subtitle01Bold">
      Box
    </Text>
    <Box {...args} />
  </>
);

Basic.args = {
  backgroundColor: 'primaryKey',
  width: 200,
  height: 200,
};
