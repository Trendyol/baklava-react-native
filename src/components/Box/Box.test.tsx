import React from 'react';
import { ReactTestInstance } from 'react-test-renderer';
import theme from '../../theme';
import { render } from '../../test-utils';
import Box from './Box';
import Text from '../Text/Text';

describe('Box', () => {
  test('should render box correctly', () => {
    // when
    const { toJSON } = render(<Box />);

    // then
    expect(toJSON()).toMatchSnapshot();
  });

  test('should render given box', () => {
    // when
    const { getByTestId } = render(
      <Box testID="box" backgroundColor="primaryKey" />,
    );

    const boxComponent = getByTestId('box');

    // then
    expect(boxComponent.props.style[0].backgroundColor).toBe(
      theme.colors.primaryKey,
    );
  });

  test('should render given box children ', () => {
    // when
    const { getByTestId } = render(
      <Box testID="box" backgroundColor="primaryKey">
        <Text>testtest</Text>
      </Box>,
    );
    const boxComponent = getByTestId('box');

    // then
    const textComponent = boxComponent.children[0] as ReactTestInstance;
    expect(textComponent.props.children).toBe('testtest');
  });
});
