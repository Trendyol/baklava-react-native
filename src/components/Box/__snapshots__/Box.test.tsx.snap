// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box should render box correctly 1`] = `
<View
  style={
    Array [
      Object {},
    ]
  }
/>
`;
